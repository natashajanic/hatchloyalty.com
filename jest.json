{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
