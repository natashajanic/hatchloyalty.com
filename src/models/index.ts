import { FluidObject } from 'gatsby-image'

export interface IBlogPost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    author: string
    date: string
    featuredImage?: IGraphQLImage
    tags: string[]
    title: string
  }
  html: string
}

export interface IGraphQLImage {
  childImageSharp: {
    fluid: FluidObject
  }
}

export interface IRelease {
  changes: Array<{ component: string; description: string; link?: string }>
  date: string
  name: string
}

export interface IResourcePost {
  excerpt: string
  fields: {
    slug: string
  }
  frontmatter: {
    title: string
  }
  html: string
}

export interface ITeamMember {
  headshot?: IGraphQLImage
  name: string
  position: string
}
