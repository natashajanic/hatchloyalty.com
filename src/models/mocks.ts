import { IRelease, IBlogPost, IResourcePost, ITeamMember } from '.'

export const mockBlogPost: IBlogPost = {
  excerpt: 'This is the excerpt for a blog post.',
  fields: {
    slug: '/2019-02-25-test-blog-post/',
  },
  frontmatter: {
    author: 'John Doe',
    date: 'February 25, 2019',
    featuredImage: {
      childImageSharp: {
        fluid: {
          aspectRatio: 1,
          src: 'test/imgs/test-blog-post.png',
          srcSet: '',
          sizes: '',
        },
      },
    },
    tags: ['Test', 'Press Release'],
    title: 'Test Blog Post',
  },
  html: '<div>Test blog post body content.</div>',
}

export const mockRelease: IRelease = {
  changes: [
    { component: 'Test', description: 'Test Change 1' },
    {
      component: 'Test HQ',
      description: 'Test Change 2',
      link: 'https://developer.hatchloyalty.com/test/change-2',
    },
  ],
  date: 'January 2019',
  name: 'Test Release 2019.01',
}

export const mockResourcePost: IResourcePost = {
  excerpt: 'This is the excerpt for a resource post.',
  fields: {
    slug: '/test-resource-post/',
  },
  frontmatter: {
    title: 'Test Resource Post',
  },
  html: '<div>Test resource post body content.</div>',
}

export const mockTeamMember: ITeamMember = {
  headshot: {
    childImageSharp: {
      fluid: {
        aspectRatio: 1,
        src: 'test/imgs/john-doe.png',
        srcSet: '',
        sizes: '',
      },
    },
  },
  name: 'John Doe',
  position: 'CEO',
}
