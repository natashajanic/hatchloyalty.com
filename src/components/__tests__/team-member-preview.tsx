import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import Img from 'gatsby-image'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Text from 'src/components/text'
import { mockTeamMember } from 'src/models/mocks'
import TeamMemberPreview, {
  ITeamMemberPreviewProps,
} from '../team-member-preview'

describe('TeamMemberPreview', () => {
  let props: ITeamMemberPreviewProps
  let mountedComponent: ShallowWrapper<any, any, TeamMemberPreview>

  beforeEach(() => {
    props = {
      teamMember: mockTeamMember,
      postCount: 12,
    }
    mountedComponent = shallow(<TeamMemberPreview {...props} />)
  })

  describe('render', () => {
    it('renders a link to view the full details of the team member', async () => {
      mountedComponent.setProps({
        teamMember: { ...mockTeamMember, name: 'Sally Smith' },
      })
      expect(mountedComponent.find(Link).length).toEqual(1)
      const titleLink = mountedComponent.find(Link).at(0)
      expect(titleLink.prop('to')).toEqual('/team/sally-smith/')
    })

    it("renders the team member's name", async () => {
      expect(mountedComponent.text()).toContain(mockTeamMember.name)
    })

    it("renders the team member's headshot", async () => {
      expect(mountedComponent.find(Img).length).toEqual(1)
      expect(mountedComponent.find(Img).prop('fluid')).toBeDefined()
    })

    it('renders a default headshot if none is set for the team member', async () => {
      mountedComponent.setProps({
        teamMember: { ...mockTeamMember, headshot: undefined },
      })
      expect(mountedComponent.find('img').length).toEqual(1)
    })

    it('renders the number of posts authored by the team member', async () => {
      expect(
        mountedComponent
          .find(Text)
          .last()
          .text()
      ).toEqual('(12 posts)')
    })

    it('uses proper pluralization for single post counts', async () => {
      mountedComponent.setProps({ postCount: 1 })
      expect(
        mountedComponent
          .find(Text)
          .last()
          .text()
      ).toEqual('(1 post)')
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
