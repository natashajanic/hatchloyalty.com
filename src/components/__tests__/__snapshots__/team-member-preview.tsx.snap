// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMemberPreview render matches the expected snapshot 1`] = `
<Styled(Component)
  blacklist={
    Array [
      "css",
      "flex",
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "textAlign",
      "cursor",
    ]
  }
  border="1px solid"
  borderColor="grayLight"
  borderRadius={4}
  flex={0}
  is={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "Body": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": false,
          "rules": Array [
            [Function],
            "max-height: 415px; overflow: scroll;",
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
          ],
          "is": "div",
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {},
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Footer": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "alignItems": "center",
          "blacklist": Array [
            "css",
            "alignItems",
            "border",
            "borderTop",
            "borderRight",
            "borderBottom",
            "borderLeft",
            "borderColor",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
          ],
          "borderColor": "grayLight",
          "borderTop": 1,
          "display": "flex",
          "is": "footer",
          "justifyContent": "space-between",
          "mt": 4,
          "mx": -3,
          "pt": 3,
          "px": 3,
        },
        "displayName": "Panel.Footer",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "border": [Function],
          "borderBottom": [Function],
          "borderColor": [Function],
          "borderLeft": [Function],
          "borderRight": [Function],
          "borderTop": [Function],
          "display": [Function],
          "justifyContent": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Header": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "alignItems": "center",
          "blacklist": Array [
            "css",
            "alignItems",
            "border",
            "borderTop",
            "borderRight",
            "borderBottom",
            "borderLeft",
            "borderColor",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "flex",
            "flexDirection",
          ],
          "borderBottom": 1,
          "borderColor": "grayLight",
          "display": "flex",
          "is": "header",
          "justifyContent": "space-between",
          "mb": 4,
          "mx": -3,
          "pb": 3,
          "px": 3,
        },
        "displayName": "Panel.Header",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "border": [Function],
          "borderBottom": [Function],
          "borderColor": [Function],
          "borderLeft": [Function],
          "borderRight": [Function],
          "borderTop": [Function],
          "display": [Function],
          "flex": [Function],
          "flexDirection": [Function],
          "justifyContent": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "Title": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": false,
          "rules": Array [
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
            [Function],
          ],
        },
        "defaultProps": Object {
          "blacklist": Array [
            "css",
            "color",
            "bg",
            "fontSize",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "alignItems",
            "display",
          ],
          "color": "offBlack",
          "fontSize": 4,
          "is": "h4",
          "m": 0,
        },
        "displayName": "Styled(Component)",
        "foldedComponentIds": Array [],
        "propTypes": Object {
          "alignItems": [Function],
          "bg": [Function],
          "color": [Function],
          "display": [Function],
          "fontSize": [Function],
          "m": [Function],
          "mb": [Function],
          "ml": [Function],
          "mr": [Function],
          "mt": [Function],
          "mx": [Function],
          "my": [Function],
          "p": [Function],
          "pb": [Function],
          "pl": [Function],
          "pr": [Function],
          "pt": [Function],
          "px": [Function],
          "py": [Function],
        },
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "systemComponent": true,
        "target": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-htpNat",
        "isStatic": false,
        "rules": Array [
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          "box-shadow: #dfdfdf 0 3px 10px;",
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "cursor",
        ],
        "border": "1px solid",
        "borderColor": "grayLight",
        "borderRadius": 4,
        "m": 0,
        "p": 3,
        "width": "100%",
      },
      "displayName": "Panel",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "alignItems": [Function],
        "border": [Function],
        "borderBottom": [Function],
        "borderColor": [Function],
        "borderLeft": [Function],
        "borderRadius": [Function],
        "borderRight": [Function],
        "borderTop": [Function],
        "display": [Function],
        "flex": [Function],
        "flexDirection": [Function],
        "height": [Function],
        "justifyContent": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "position": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "width": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-htpNat",
      "systemComponent": true,
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  key="john-doe"
  m={3}
  p={3}
  px={4}
  py={4}
  textAlign="center"
  width="100%"
>
  <ForwardRef
    to="/team/john-doe/"
  >
    <Styled(Component)
      blacklist={
        Array [
          "css",
          "borderRadius",
          "height",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "overflow",
          "width",
          "css",
          "alignItems",
          "border",
          "borderColor",
          "borderBottom",
          "borderRadius",
          "color",
          "bg",
          "display",
          "justifyContent",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "width",
        ]
      }
      borderRadius={50}
      height={200}
      is={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "rules": Array [
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
            ],
          },
          "defaultProps": Object {
            "blacklist": Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ],
          },
          "displayName": "Styled(Component)",
          "foldedComponentIds": Array [],
          "propTypes": Object {
            "alignItems": [Function],
            "bg": [Function],
            "border": [Function],
            "borderBottom": [Function],
            "borderColor": [Function],
            "borderRadius": [Function],
            "color": [Function],
            "display": [Function],
            "justifyContent": [Function],
            "m": [Function],
            "mb": [Function],
            "ml": [Function],
            "mr": [Function],
            "mt": [Function],
            "mx": [Function],
            "my": [Function],
            "p": [Function],
            "pb": [Function],
            "pl": [Function],
            "pr": [Function],
            "pt": [Function],
            "px": [Function],
            "py": [Function],
            "width": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "systemComponent": true,
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      mx="auto"
      my={2}
      overflow="hidden"
      width={200}
    >
      <Image
        Tag="div"
        alt=""
        critical={false}
        fadeIn={true}
        fluid={
          Object {
            "aspectRatio": 1,
            "sizes": "",
            "src": "test/imgs/john-doe.png",
            "srcSet": "",
          }
        }
      />
    </Styled(Component)>
    <Text
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      fontSize={4}
      is="h2"
      lineHeight="1.5"
      m={0}
      textStyle="normal"
    >
      John Doe
    </Text>
    <Text
      blacklist={
        Array [
          "css",
          "fontSize",
          "lineHeight",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "textStyle",
          "alignItems",
          "color",
          "bg",
          "display",
          "flex",
          "fontWeight",
          "textAlign",
        ]
      }
      fontSize={3}
      lineHeight="1.5"
      m={0}
      textStyle="normal"
    >
      (
      12
       post
      s
      )
    </Text>
  </ForwardRef>
</Styled(Component)>
`;
